File: agents/tabos/AgentNodeMain.py
import os
import sys
import asyncio
import nest_asyncio
from AgentNode import AgentNode

nest_asyncio.apply()

pathLib = os.path

getArrayItem = lambda arr, index: arr[index] if 0 <= index < len(arr) else None

# Get current path:
current_path = cwd=os.getcwd() # pathLib.dirname(pathLib.realpath(__file__))
print("Run path:", current_path)
print("System path: ",sys.path)

# argv[1]: agentPath, argv[2]: host address, argv[3]: host name
agentPath=getArrayItem(sys.argv,1)
host=getArrayItem(sys.argv,2)
name=getArrayItem(sys.argv,3)
agentNode = AgentNode(agentPath, host , name)

# Set up an async looper for async calls:
loop = asyncio.get_event_loop()
loop.set_debug(True)  # 启用调试
loop.run_until_complete(agentNode.start(True))

File: AgentNodeMain.mjs
import dotEnv from "dotenv";
import pathLib from "path";
import AgentNode from "./agenthub/AgentNode.mjs";

let agentPath=process.argv[2];
let hostAddr=process.argv[3];
let name= process.argv[4];

let envFileName=process.argv[5];
if(envFileName){
	if(envFileName.indexOf("/")>=0){
		dotEnv.config({path:envFileName});
	}else{
		dotEnv.config({path:"./"+envFileName});
	}
}else {
	dotEnv.config();
}
await AgentNode.setupPath();
let agentNode=new AgentNode(agentPath,hostAddr,name);
agentNode.start();


